l10n_nl_xaf_auditfile_export/models/xaf_auditfile_export.py

```diff
@@ -251,6 +251,48 @@ def get_taxes(self):
             ('company_id', '=', self.company_id.id),
         ])
 
+    @api.multi
+    def get_ob_totals(self):
+        """return totals of opening balance"""
+        self.env.cr.execute(
+            "select sum(l.credit), sum(l.debit), count(distinct a.id) "
```

Comment:
Sorry if I missed this in any earlier version I might have reviewed, but this seems to greatly overinflate the numbers in my test setup.

If I receive 1000 euros on my bank account before the start date, and I pay 1000 euros from my bank account before the start date, this 1000 euros should not be included in the summary of the opening balance but here it is included on both the debit and the credit side. Therefore, while the opening balance entries are correct in the exported XML, the totals are not (unless I should be reading the specifications to see that I misunderstand the meaning of these numbers).

Reply:
@thomaspaulb did you dive into this?

Reply:
@StefanRijnhart might well be right about this, the situation I encountered before adding this was that the opening balances were totally missing from the XAF, which is what I sought to correct with minimal understanding of the issue. I haven't had complaints about it from clients, but that doesn't mean there couldn't be a problem here. @gjotten could you perhaps advise from a more accounting-informed perspective?

Reply:
@StefanRijnhart do you mean that we're at least finding the correct accounts we should be posting an opening balance for, but not the correct balance amounts on them?

Intuitively and in a general sense, thinking that +1000 -1000 should even out into a 0.00 total amount (i.e. balance) at the start of the year would be correct. I can also imagine that such a 0.00 balance on an account should not even trigger posting an entry in the .xaf.

I gather that the .xaf specifications also allow us to get away with simply _either_ having a total amount positive or negative (debit or credit) posted as an amount (=balance) on a single account, but not 0/00?

Are we doing balance calculation somewhere? @thomaspaulb I don't think I see it, but my Pythonese is limited.

Reply:
If you simply group the balances retrieved in `get_ob_lines` by sign (debit/credit) and sum them you will get the right results.

Reply:
To at least translate the Pythonese, or SQL-ese: what the code is doing is:

- taking any account that according to its account type is supposed to be part of the opening balance
- summing debit and credit for all account move lines prior to the date upon which the opening balance is requested, and including this as TotalDebit and TotalCredit
- summing the balance per account for each move line, and including each of these under `obLines`

Looking now at these [mediocre, second-hand docs](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_1024093953.html#subsect_60094306057) I think what I should have done is:

- taking any account that according to its account type is supposed to be part of the opening balance
- summing the balance per account for each move line, and including each of these under `obLines`
- for the above obLines, check out whether the balance comes out as debit or credit; sum all the debit balances and all the credit balances separately, and include the sums as TotalDebit and TotalCredit

This will lead to way less inflated numbers

Reply:
... what Stefan says is what I mean, but more concise :)

Reply:
I'll come up with a version of this tonight, thanks everyone

Reply:
@gjotten @StefanRijnhart @thomaspaulb now I calculate the totals from the same set as the lines, and filter for posted moves too.

I have a history of getting a knot in my brain from the balance sign of different types of accounts, so please be extra critical there